packages/thirdweb/src/wallets/utils/convertAccount.ts

```diff
@@ -0,0 +1,23 @@
+import type { Signer } from "ethers6";
+import type { WalletClient } from "viem";
+import { ethers6Adapter } from "../../adapters/ethers6.js";
+import { viemAdapter } from "../../adapters/viem.js";
+import type { Account, AnyAccount } from "../interfaces/wallet.js";
+
+function isViem(anyAccount: AnyAccount): anyAccount is WalletClient {
+  return "account" in anyAccount;
+}
+
+function isEthers6(anyAccount: AnyAccount): anyAccount is Signer {
+  return "provider" in anyAccount;
```

Comment:
Can we make it work with ethers5 too?

Reply:
Any specific reason why? ethers 5 has all kinds of discrepancies that make things difficult